server/etcdserver/server.go

```diff
@@ -1072,7 +1072,12 @@ func (s *EtcdServer) applySnapshot(ep *etcdProgress, apply *apply) {
 
 	// As backends and implementations like alarmsStore changed, we need
 	// to re-bootstrap Appliers.
-	s.uberApply = newUberApplier(s)
+	s.uberApply = s.NewUberApplier()
+}
+
+func (s *EtcdServer) NewUberApplier() *uberApplier {
+	return newUberApplier(s.lg, s.be, s.KV(), s.alarmStore, s.authStore, s.lessor, s.cluster, s, s, s.consistIndex,
```

Comment:
Are we sure that during runtime (after bootstrap) none of those fields on server are recreated? For example restoring snapshot, recovering auth etcd?

Reply:
They are in the `applySnapshot` and this is handled.  I haven't found other places were we 'recreate' them.